cou/utils/openstack.py

```diff
@@ -214,14 +285,27 @@ def next_release(self) -> Optional[str]:
         """Return the next OpenStack release codename.
 
         :return: OpenStack release codename.
-        :rtype: str
+        :rtype: Optional[str]
         """
         try:
             return self.openstack_codenames[self.index + 1]
         except IndexError:
-            logger.warning("There is no OpenStack release after %s", self.codename)
+            logger.warning("Cannot find an OpenStack release after %s", self.codename)
             return None
 
+    @property
+    def previous_release(self) -> Optional[str]:
+        """Return the previous OpenStack release codename.
+
+        :return: OpenStack release codename.
+        :rtype: Optional[str]
+        """
+        if self.index == 0:
```

Comment:
using the EAFP style would better align with the `next_release` property

Reply:
The problem of using `EAFP` on `previous_release` is that an index -1 is possible and would result into the latest OpenStack release. E.g: you ask for the previous release of `diablo` and that would result in `caracal`. There is no `IndexError` for negative indexes.

Reply:
duh of course... :clown_face: :facepalm: